cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskRetentionManager.java

```diff
@@ -0,0 +1,214 @@
+                                 new RetentionInfo(ADMIN_CRUISE_CONTROL.endPoints(),
+                                                   config == null ? defaultUserTaskRetentionTimeMs
+                                                                  : config.getLong(KafkaCruiseControlConfig.
+                                                                                       COMPLETED_ADMIN_USER_TASK_RETENTION_TIME_MS_CONFIG),
+                                                   config == null ? defaultUserTaskRetentionLimit
+                                                                  : config.getInt(KafkaCruiseControlConfig.
+                                                                                      MAX_CACHED_COMPLETED_ADMIN_USER_TASKS_CONFIG)));
+    _time = time;
+  }
+
+  /**
+   * Add {@link UUID} associated with a UserTask to corresponding category in {@link UserTaskRetentionManager}
+   * @param uuid {@link UUID} to be added
+   * @param endPoint {@link EndPoint} used to determine category of UserTask
+   */
+  public void addUserTaskUUID(UUID uuid, EndPoint endPoint) {
+    RetentionInfo retentionInfo = retentionInfoFromEndPoint(endPoint);
+    if (retentionInfo != null) {
+      retentionInfo.addUUID(uuid);
+    }
+  }
+
+  private RetentionInfo retentionInfoFromEndPoint(EndPoint endPoint) {
+    for (RetentionInfo retentionInfo : _endPointRetentionByType.values()) {
+      if (retentionInfo.endPoints().contains(endPoint)) {
+        return retentionInfo;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Removes UserTasks for a certain {@link RetentionType} that has exceeded retention time for that type.
+   * @param idToUserTask Map of @{link UUID} to UserTasks
+   */
+  public void removeOldUserTasks(Map<UUID, UserTaskManager.UserTaskInfo> idToUserTask) {
+    idToUserTask.entrySet().removeIf(entry -> {
+      UserTaskManager.UserTaskInfo userTaskInfo = entry.getValue();
+      return userTaskInfo.startMs() + retentionInfoFromEndPoint(userTaskInfo.endPoint()).timeMs() < _time.milliseconds();
+    });
+  }
+
+  /**
+   * Removes UserTasks for a certain {@link RetentionType} that has exceeded count limit for that type. Remove is in
+   * FIFO order.
+   * @param idToUserTask Map of @{link UUID} to UserTasks
+   */
+  public void removeExtraUserTasks(Map<UUID, UserTaskManager.UserTaskInfo> idToUserTask) {
+    for (RetentionInfo retentionInfo : _endPointRetentionByType.values()) {
+      UUID uuid = retentionInfo.getUUIDToRemove();
+      while (uuid != null) {
+        idToUserTask.remove(uuid);
+        uuid = retentionInfo.getUUIDToRemove();
+      }
+    }
+  }
+
+  /**
+   * A class to define retention policy to keep track of retention parameters for one semantic category, as well as list
+   * of {@link UUID} for UserTasks being retained under this category.
+   */
+  private static class RetentionInfo {
+    private final Set<EndPoint> _endPoints;
+    private final long _timeMs;
```

Comment:
Can you add a comment to describe `_timeMs` and `_limit` here? Or maybe rename them to be more descriptive.